beacon_node/beacon_chain/src/beacon_chain_builder.rs

```diff
@@ -210,22 +210,20 @@ fn interop_genesis_state<T: EthSpec>(
         .collect::<Vec<_>>();
 
     let mut proofs = vec![];
+    let depth = spec.deposit_contract_tree_depth as usize;
+    let mut tree = MerkleTree::create(&[], depth);
     for i in 1..=deposit_root_leaves.len() {
-        // Note: this implementation is not so efficient.
-        //
-        // If `MerkleTree` had a push method, we could just build one tree and sample it instead of
-        // rebuilding the tree for each deposit.
-        let tree = MerkleTree::create(
-            &deposit_root_leaves[0..i],
-            spec.deposit_contract_tree_depth as usize,
-        );
+        match tree.push_leaf(deposit_root_leaves[i], depth) {
```

Comment:
This loop will crash on the last index (`1..=deposit_root_leaves.len()`). Could rewrite it in terms of `deposit_root_leaves.iter()`

Reply:
The match could also be an `if let`

Reply:
Used .iter().enumerate() for proof generation. Thanks for your help :)